walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -26,38 +26,73 @@
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     List<TimeRange> possibleTimes = new ArrayList<TimeRange>();
     Set<String> attendees =  new HashSet<String>();
+    Set<String> optionalAttendees =  new HashSet<String>();
     int meetingDur = (int) request.getDuration();
     attendees.addAll(request.getAttendees());
+    optionalAttendees.addAll(request.getOptionalAttendees());
     if (meetingDur > TimeRange.WHOLE_DAY.duration()){
       return possibleTimes;
     }
-    if (events.isEmpty() || attendees.isEmpty()){
+    if (events.isEmpty()){
       possibleTimes.add(TimeRange.WHOLE_DAY);
       return possibleTimes;
     }
-    ArrayList<TimeRange> busyWithOverlap = getBusyTimes(events, attendees);
-    possibleTimes = findFreeTimes(busyWithOverlap, request.getDuration());
+    possibleTimes = getPossibleTimes(events, attendees, optionalAttendees, request.getDuration());
     return possibleTimes;
   }
 
 /**
-  * Returns an ArrayList of meeting times that are unavailable due to schedule
-  * conflicts and overlapping meeting times within the given group of attendees. 
+  * Returns an ArrayList of possible meeting times that take into account mandatory
```

Comment:
I guess what's confusing here, is exactly what it returns.  Good documentation should make clear exactly what happens.

I mean are optional attendees times treated differently in any ways when computing these times?

Reply:
Breaking into smaller functions (hopefully) helped clarify the specifics of what the functions do, javadocs were updated to reflect that